### PR TITLE
Fixed adding or removing contacts to campaigns when they are added or removed to/from segments

### PR DESCRIPTION
**Pull request by alanhartless**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
https://github.com/mautic/mautic/pull/6963 added support for "view other" campaign permissions. But it added the permission check to code that manipulated contacts' campaign membership based on segment membership. This code is mainly executed in circumstances outside a user session (cli, form submit, etc) and has no effect anyway considering the `mautic:campaign:update` command results in the same outcome (contact is added or removed). 

Finding campaigns associated with a segment for these use cases should not be restricted to whether the user has campaign access or not since those that do have access expect contacts to be added or removed from their campaign when the contact is added or removed from the segment they associated with the segment. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a new segment with a filter
2. Create a new campaign with the segment as a source
3. Update a contact to match the filter
4. Run the command `php bin/console mautic:segments:update`
5. Notice the contact will be added to the segment but NOT the campaign

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
1. Create a new segment with a filter
2. Create a new campaign with the segment as a source
3. Update a contact to match the filter
5. Notice the contact will be added to the segment and the campaign

### CODE REVIEW

**Review by pavel-janicek — DISMISSED** [review]
tested on another issue - #7640  and this PR fixes the issue. Approved

**Comment by RCheesley** [comment]
Looks like this one is failing on Travis @alanhartless so may need some attention when you're back :)

**Comment by dennisameling** [comment]
Let me close and re-open this PR since there are `Constant MAUTIC_INSTALLER already defined` errors in Travis that I think shouldn't have to be there. Let's see what happens if Travis runs again, so that we have an up-to-date list of errors that need fixing.

**Comment by npracht** [comment]
Hi there!
It has been decided to not create any extra Mautic 2 versions (except for major security purpose) knowing that Mautic 3 is coming very soon.

**We now want to integrate your contribution in the Mautic 3 roadmap as 3.0.1 candidate.**

## How to do?
1. Check if the bug still present in Mautic 3
2. **If Yes**: rebase your PR against the 3.x branch
3. **If No**: please comment on your PR: “This PR only applies to Mautic 2.x”

**Please report results by commenting on your PR to make us administration easier.**

_In case your bugfix only apply to Mautic 2, we'll consider adding it in an extra Mautic 2 version._
______
You can more information on how to do all of that on [this blog post "Getting you PR ready for Mautic 3"](https://www.mautic.org/blog/developer/getting-your-pr-ready-mautic-3).

**Comment by npracht** [comment]
Hello @alanhartless could you please test mautic 3.0.0-beta2 and tell me if the issue is still existing?
- If no, i'll tag this issue at `M2 only relevant`
- If yes, could you please rebase your branch against 3.x branch ?

Waiting for your feedback, i would very love to merge it in 3.0.1 if still relevant. Thanks !

**Comment by RCheesley** [comment]
@alanhartless please see conflicts to be resolved

**Comment by RCheesley** [comment]
Bumping to 3.1.1 due to unresolved conflicts

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
Bumping to 3.1.2 due to unresolved conflicts, @mautic/acquia-po this needs to be addressed by **18th October**

**Comment by dennisameling** [comment]
Bumping to 3.2 due to unresolved conflicts. CC @mautic/acquia-po

**Comment by alanhartless** [comment]
Conflicts resolved

**Review by anton-vlasenko — DISMISSED** [review]
I've code reviewed  and tested this task.
I followed "Steps to reproduce" and my test contact was added to a campaign.
Screen Shot 2021-02-04 at 12 26 44 PM
Therefore I approve this PR.

**Comment by anton-vlasenko** [comment]
I think this PR can be merged because it has 2 CR approvals.
I've only added a unit test and it doesn't affect the logic.

**Comment by RCheesley** [comment]
@anton-vlasenko can you please rebase against the features branch then we should be good to go.

**Comment by anton-vlasenko** [comment]
@RCheesley I've rebased the PR.

**Review by RCheesley — APPROVED** [review]

**Pull request merged**
